### PR TITLE
Shift SearchTimeoutTests into core tests minus the Groovy dependency.

### PR DESCRIPTION
**Pull request by markharwood**
Removed the Groovy dependency for testing searches that timeout are flagged. Can now move back into core away from lang-groovy plugin

### CODE REVIEW

File: core/src/test/java/org/elasticsearch/search/SearchTimeoutTests.java [hunk 1]
```diff
@@ -36,12 +41,12 @@
 
 /**
  */
-@ESIntegTestCase.ClusterScope(scope= ESIntegTestCase.Scope.SUITE)
+@ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.SUITE)
 public class SearchTimeoutTests extends ESIntegTestCase {
```
Comment (jpountz) [inline core/src/test/java/org/elasticsearch/search/SearchTimeoutTests.java]: this needs to be called SearchTimeoutIT in core

**Comment by jpountz** [comment]
LGTM, just a minor change to perform to the test name

**Comment by markharwood** [comment]
Pushed to master in https://github.com/elastic/elasticsearch/commit/c3a50d7ca2c507327c670614e01b54add9231469

Still working on the backport to 2.x due to eclipse IDE setup issues

**Comment by markharwood** [comment]
Backported to 2.x with https://github.com/elastic/elasticsearch/commit/d0f3b41b667140437a5d5fc96fb819b37e69c485

**Pull request closed without merging**